### ISSUE TITLE
Problems with image caching

### ISSUE THREAD

**Issue by robertlight** [openseadragon/openseadragon#1427]
I am a relative newbie with OSD.

When I regenerate the underlying image, I have to clear the browser's cache so that I can load in the new tiles.  Otherwise all I see are the old image tiles.

Is there a way to tell OSD that it should treat all tiles as if they must be reloaded from the server "anew".

If I set up the server to add no-cache headers to the images, it solves my problem but with a performance hit because then OSD never-ever caches the tiles.

In other words... I want the tiles cached as the user browses around the image but on a new page load, I want all tiles treated as if they have to be drawn from the server the first time.

If I was writing OSD, I would allow the user to add a query suffix to the image such as:
?timestamp=12345

so that all images get loaded the first time and then cached from that point on.

Many thanks for help !!

**Comment by Kampii** [comment]
Joining the party.. :)

We also had the same problem that we never were able to figure out. I think it would be ideal if it was possible to cache the tiles from all the different images that are being loaded by passing, say imgId=1, imgId=2 etc, AND at the same time be able to redraw the tiles of the new image id correctly AND clear the tile from the previous id. We had tried viewer.world.resetItems() ([related thread](https://github.com/openseadragon/openseadragon/issues/866)) in the past, but somehow were not able to get it to work and ended up having "tile failed to load" in console. I am also curious what is the solution here.

**Comment by robertlight** [comment]
I think it would be desirable to have the setup for OSD to be like:
`<script type="text/javascript">
    var viewer = OpenSeadragon({
        id: "openseadragon1",
        prefixUrl: "/openseadragon/images/",
        tileSources: "/path/to/my/image.dzi",
        tlleSuffix: "?page-load-timestamp"
    });
</script>`

This way all the tile images would be loaded from:  
`/path/to/my/image_files/..../tile-image.png?page-load-timestamp`

Since the tag and the timestamp are the same as you pan & zoom over your image, the tiles will be cached naturally... but if the page gets reloaded, then all the tiles will be fetch anew the first time they are needed...but not again as you pan & zoom.

This way if the tiles get regenerated (as they do in our case) they load afresh the first time but as you pan and zoom, they don't have to be reloaded.

**Comment by iangilman** [comment]
Actually I believe it already works, like so: 

```
<script type="text/javascript">
     var viewer = OpenSeadragon({
         id: "openseadragon1",
         prefixUrl: "/openseadragon/images/",
         tileSources: "/path/to/my/image.dzi?page-load-timestamp"
     });
 </script> 
```

It'll automatically add `?page-load-timestamp` to each tile request.

I suppose that feature isn't well documented... Not sure what the best place would be, but it would be a good idea to document it somewhere!

**Comment by robertlight** [comment]
Thank you !!!  Just what I needed.
